src/sacn/private/dmx_merger.h

```diff
@@ -29,14 +29,63 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include "../../../include/sacn/dmx_merger.h"
+#include "sacn/private/util.h"
+#include "etcpal/rbtree.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef struct SourceState
+{
+  sacn_source_id_t handle;
+  SacnDmxMergerSource source;
+} SourceState;
+
+typedef struct CidToSourceHandle
+{
+  EtcPalUuid cid;
+  sacn_source_id_t handle;
+} CidToSourceHandle;
+
+typedef struct MergerState
+{
+  sacn_dmx_merger_t handle;
+
+  IntHandleManager source_handle_mgr;
+
+  EtcPalRbTree source_state_lookup;
+  EtcPalRbTree source_handle_lookup;
+
+  const SacnDmxMergerConfig* config;
+  uint8_t winning_priorities[DMX_ADDRESS_COUNT];
+} MergerState;
+
+IntHandleManager merger_handle_mgr;
+EtcPalRbTree mergers;
```

Comment:
Don't define global variables in header files. If a global variable REALLY MUST be accessible outside a single .c file, declare it as `extern` in the header file and define it in the .c file. Otherwise define it as `static` in the .c file.